x/crosschain/types/cmd_cctxs.go

```diff
@@ -0,0 +1,244 @@
+package types
+
+import (
+	"fmt"
+
+	errorsmod "cosmossdk.io/errors"
+	sdkmath "cosmossdk.io/math"
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+
+	"github.com/zeta-chain/zetacore/pkg/chains"
+	"github.com/zeta-chain/zetacore/pkg/coin"
+	"github.com/zeta-chain/zetacore/pkg/constant"
+	zetacrypto "github.com/zeta-chain/zetacore/pkg/crypto"
+	"github.com/zeta-chain/zetacore/pkg/gas"
+)
+
+// MigrateERC20CustodyFundsCmdCCTX returns a CCTX allowing to migrate ERC20 custody funds
+func MigrateERC20CustodyFundsCmdCCTX(
+	creator string,
+	erc20Address string,
+	custodyContractAddress string,
+	newCustodyContractAddress string,
+	chainID int64,
+	amount sdkmath.Uint,
+	gasPrice string,
+	priorityFee string,
+	tssPubKey string,
+	currentNonce uint64,
+) CrossChainTx {
+	indexString := GetERC20CustodyMigrationCCTXIndexString(tssPubKey, currentNonce, erc20Address)
+	hash := crypto.Keccak256Hash([]byte(indexString))
+
+	return newCmdCCTX(
```

Comment:
Do we really need this amount of arguments passed to a single function?

Reply:
Can be refactored but not it allow to group cmd cctx initialization in a single place